### PR TITLE
FINERACT-1926 storing balances at ownership transfer

### PR DESCRIPTION
**Pull request by reluxa**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-investor/src/main/java/org/apache/fineract/investor/domain/ExternalAssetOwnerTransferDetails.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    @Column(name = "interest_outstanding_derived", scale = 6, precision = 19, nullable = false)
+    private BigDecimal totalInterestOutstanding;
+
+    @Column(name = "fee_charges_outstanding_derived", scale = 6, precision = 19, nullable = false)
+    private BigDecimal totalFeeChargesOutstanding;
+
+    @Column(name = "penalty_charges_outstanding_derived", scale = 6, precision = 19, nullable = false)
+    private BigDecimal totalPenaltyChargesOutstanding;
+
+    @Column(name = "total_overpaid_derived", scale = 6, precision = 19)
```
Comment (adamsaghy) [inline fineract-investor/src/main/java/org/apache/fineract/investor/domain/ExternalAssetOwnerTransferDetails.java]: is there a reason this one can be null? (above fields are all not nullable)

File: fineract-investor/src/main/resources/db/changelog/tenant/module/investor/parts/0007_add_external_asset_owner_transfer_details.xml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            <column name="total_overpaid_derived" type="DECIMAL(19, 6)">
+                <constraints nullable="false"/>
+            </column>
+
+            <column name="created_by" type="BIGINT" />
+            <column name="created_on_utc" type="TIMESTAMP WITH TIME ZONE" />
+            <column name="last_modified_by" type="BIGINT" />
+            <column name="last_modified_on_utc" type="TIMESTAMP WITH TIME ZONE" />
+        </createTable>
+    </changeSet>
```
Comment (adamsaghy) [inline fineract-investor/src/main/resources/db/changelog/tenant/module/investor/parts/0007_add_external_asset_owner_transfer_details.xml:63]: please dont forget the changeset for mysql/mariadb

File: fineract-investor/src/main/java/org/apache/fineract/investor/domain/ExternalAssetOwnerTransferDetails.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import lombok.Setter;
+import org.apache.fineract.infrastructure.core.domain.AbstractAuditableWithUTCDateTimeCustom;
+
+@Getter
+@Setter
+@Table(name = "m_external_asset_owner_transfer_details")
+@NoArgsConstructor
+@Entity
+public class ExternalAssetOwnerTransferDetails extends AbstractAuditableWithUTCDateTimeCustom {
+
```
Comment (adamsaghy) [inline fineract-investor/src/main/java/org/apache/fineract/investor/domain/ExternalAssetOwnerTransferDetails.java:39]: Please dont forget the tests. Feel free to enhance the existing ones:
`InitiateExternalAssetOwnerTransferTest`

**Review by adamsaghy — CHANGES_REQUESTED** [review]
Please see my comments!

File: fineract-investor/src/main/java/org/apache/fineract/investor/domain/ExternalAssetOwnerTransferDetails.java (hunk 1 above)
Comment (reluxa) [reply fineract-investor/src/main/java/org/apache/fineract/investor/domain/ExternalAssetOwnerTransferDetails.java]: done

File: fineract-investor/src/main/resources/db/changelog/tenant/module/investor/parts/0007_add_external_asset_owner_transfer_details.xml (hunk 2 above)
Comment (reluxa) [reply fineract-investor/src/main/resources/db/changelog/tenant/module/investor/parts/0007_add_external_asset_owner_transfer_details.xml:63]: added

File: fineract-investor/src/main/java/org/apache/fineract/investor/domain/ExternalAssetOwnerTransferDetails.java (hunk 3 above)
Comment (reluxa) [reply fineract-investor/src/main/java/org/apache/fineract/investor/domain/ExternalAssetOwnerTransferDetails.java:39]: done

**Review by adamsaghy — APPROVED** [review]
LGTM

**Pull request merged**